lib/stdlib/src/edlin_expand.erl

```diff
@@ -223,3 +227,19 @@ over_white([$\t|Cs], Stack, N) ->
     over_white(Cs, [$\t|Stack], N+1);
 over_white(Cs, Stack, N) when is_list(Cs) ->
     {Cs,Stack,N}.
+
+get_exports(Mod) 
+	when is_atom(Mod) -> get_exports(erlang:atom_to_list(Mod));
+
+get_exports(Mod)
+	when is_list(Mod) -> 
+	case beam_lib:chunks(code:where_is_file(Mod ++ ".beam"), [exports]) of
+		{ok, {_,E}} -> E ;
+		{error, _, _} -> []
+	end.
+
+find_modules() ->
```

Comment:
What we do in Elixir, and it may be more performant and accurate, is that we call `application:loaded_applications()` to get all loaded apps, then we call `application:get_key(App, modules)` to get all modules for that application.

It requires the applications to be loaded before, but they usually are, including in rebar's case.

Reply:
Here is the relevant source: https://github.com/elixir-lang/elixir/blob/4ad5749d00b2e607a976feb1445b1635ff99f1dc/lib/iex/lib/iex/autocomplete.ex#L393-L408

Reply:
`make shell` is just an interactive shell with the Erlang.mk environment and no particular application is loaded. It's basically just calling `erl`. Your approach wouldn't work in that case, and would also not work when just working on modules in the current working directory. Not all Erlang code is found within an application when working with the shell.

Reply:
but my approach list all the modules in code path, not only application, but also simple library.

Reply:
anyway I need to read Beam file content for exports, without loading anything.

Reply:
Ah, good point @essen!

Reply:
I would recommend Elixir doing like me then ;>)